### PR TITLE
Creation of AutoLaws record and extentension of ShowTheoryLong

### PR DESCRIPTION
**Pull request by saqzari**
I haven't implemented the requested AbstractUI changes concerning looking up inputted laws yet but have completed the changes relating to AutoLaws record and showtheoryLong

### CODE REVIEW

File: src/Classifier.lhs [hunk 1]
```diff
@@ -5,6 +5,7 @@ Copyright  Saqib Zardari (c) 2023
 LICENSE: BSD3, see file LICENSE at reasonEq root
 \end{verbatim}
 \begin{code}
+{-# LANGUAGE RecordWildCards #-}
```
Comment (andrewbutterfield) [inline src/Classifier.lhs]: why this? should be necessary?

File: src/Classifier.lhs [hunk 2]
```diff
@@ -21,6 +22,14 @@ data AutoLaws = AutoLaws
   }
   deriving (Eq,Show,Read)
 
+showSimpStr :: [(String, Direction)] -> String
+showSimpStr (x:[]) = fst x
```
Comment (andrewbutterfield) [inline src/Classifier.lhs]: you want to show both string and direction, not just the strings

File: src/Classifier.lhs [hunk 3]
```diff
@@ -21,6 +22,14 @@ data AutoLaws = AutoLaws
...
+showSimpStr (x:xs) = fst x ++ showSimpStr xs
+
+showAuto AutoLaws{..} =  "simps: "   ++ showSimpStr simps  ++ "\n"
```
Comment (andrewbutterfield) [inline src/Classifier.lhs]: What is this?  Is this what requires `RecordWildCards` ?

File: src/Classifier.lhs (hunk 1 above)
Comment (saqzari) [reply src/Classifier.lhs]: I think it was needed for the syntax I used for printing the autoLaws

File: src/Classifier.lhs (hunk 3 above)
Comment (andrewbutterfield) [reply src/Classifier.lhs]: Use code like

```
showAuto  alaws = "simps: " ++ showSimStr (simps alaws) ++ "\n"
```
Use the field name `simps` to access the field

**Comment by saqzari** [comment]
Made another commit to include fixes to "show" to include direction and also added in the requested AbstractUI changes, looking up the law and taking out the term

**Review by andrewbutterfield — CHANGES_REQUESTED** [review]
the program crashes for me when I try to view a theory - crashes with misisng field

Have you  compiled  *and* run the code (in devel mode)?

```
$ req -d
starting REPL...
Welcome to the reasonEq 0.8.0.0 REPL
Type '?' for help.

🛠 devproj.Equiv> sh T
Theory 'Equiv'
Knowns:
≡ : (𝔹 ⟶ (𝔹 ⟶ 𝔹))
Laws:
   1. ⊤  “true”      true  ⊤
   2. ⊤  “≡_refl”    P  ≡  P  ⊤
   3. ⊤  “≡_assoc”   ((P≡Q)≡R)  ≡  (P≡(Q≡R))  ⊤
   4. ⊤  “≡_symm”    P≡Q≡Q≡P  ⊤
   5. ⊤  “id_subst”  P[x$/x$]  ≡  P  ⊤
Conjectures:
   1. ❓  “≡_id”        (true≡Q)  ≡  Q  ⊤
   2. ❓  “true_subst”  true[e$/x$]  ≡  true  ⊤
   3. ❓  “≡_subst”     (P≡Q)[e$/x$]  ≡  (P[e$/x$]≡Q[e$/x$])  ⊤
AutoLaws:
req: src/Theories.lhs:(115,5)-(121,12): Missing field in record construction auto
```

**Comment by saqzari** [comment]
> the program crashes for me when I try to view a theory - crashes with misisng field
> 
> Have you compiled _and_ run the code (in devel mode)?
> 
> ```
> $ req -d
> starting REPL...
> Welcome to the reasonEq 0.8.0.0 REPL
> Type '?' for help.
> 
> 🛠 devproj.Equiv> sh T
> Theory 'Equiv'
> Knowns:
> ≡ : (𝔹 ⟶ (𝔹 ⟶ 𝔹))
> Laws:
>    1. ⊤  “true”      true  ⊤
>    2. ⊤  “≡_refl”    P  ≡  P  ⊤
>    3. ⊤  “≡_assoc”   ((P≡Q)≡R)  ≡  (P≡(Q≡R))  ⊤
>    4. ⊤  “≡_symm”    P≡Q≡Q≡P  ⊤
>    5. ⊤  “id_subst”  P[x$/x$]  ≡  P  ⊤
> Conjectures:
>    1. ❓  “≡_id”        (true≡Q)  ≡  Q  ⊤
>    2. ❓  “true_subst”  true[e$/x$]  ≡  true  ⊤
>    3. ❓  “≡_subst”     (P≡Q)[e$/x$]  ≡  (P[e$/x$]≡Q[e$/x$])  ⊤
> AutoLaws:
> req: src/Theories.lhs:(115,5)-(121,12): Missing field in record construction auto
> ```

Isn't that due to the warnings you mentioned not to touch for now?

Yes - but I didn't understand what happens at runtime - you need to create a `nullAutoLaws` and use that to intialise `auto` in `nullTheory` and `readTheory` in the `Theories` module. Other wise we can't look at a theory

**Comment by saqzari** [comment]
The issues with printing theories should be fixed now. Please let me know if any other crashes come up

**Review by andrewbutterfield — APPROVED** [review]
This now works.
I'll merge this to main. That should bring in changes I made ot `Sequents.lhs` there.
You can then merge those changes into your `auto` branch

**Pull request merged**